mule-deployer/src/main/java/org/mule/tools/client/cloudhub/CloudHubClient.java

```diff
@@ -65,157 +66,134 @@ public CloudHubClient(CloudHubDeployment cloudhubDeployment, DeployerLog log) {
     super(cloudhubDeployment, log);
   }
 
-  public Application createApplication(ApplicationMetadata metadata) {
-    Entity<String> json = createApplicationRequest(metadata);
-    Response response = post(baseUri, APPLICATIONS_PATH, json);
-    if (response.getStatus() == CREATED) {
-      return response.readEntity(Application.class);
+  public List<Application> getApplications() {
+    Response response = get(baseUri, APPLICATIONS_PATH);
+
+    if (response.getStatus() == OK) {
+      return response.readEntity(new GenericType<List<Application>>() {});
     } else {
       throw new ClientException(response);
     }
   }
 
-  private Entity<String> createApplicationRequest(ApplicationMetadata metadata) {
-    String json = String.format(CREATE_REQUEST_TEMPLATE, metadata.getName(), metadata.getRegion(),
-                                metadata.getMuleVersion().get(), metadata.getWorkers(),
-                                metadata.getWorkerType());
-    json = addProperties(metadata.getProperties(), json);
-    json = json + "}";
-    return Entity.json(json);
-  }
+  public Application getApplication(String domain) {
+    checkArgument(isNotBlank(domain), "The domain must not be null nor empty.");
 
-  private Entity<String> updateApplicationRequest(ApplicationMetadata metadata) {
-    String json =
-        String.format(UPDATE_REQUEST_TEMPLATE, metadata.getRegion(), metadata.getMuleVersion().get(), metadata.getWorkers(),
-                      metadata.getWorkerType());
-    json = addProperties(metadata.getProperties(), json);
-    json = json + "}";
-    return Entity.json(json);
-  }
-
-  private String addProperties(Map<String, String> properties, String json) {
-    if (properties == null) {
-      return json;
-    }
+    Response response = get(baseUri, format(A_APPLICATION_PATH, domain));
 
-    json = json + ",";
-    json = json + "  \"properties\": {";
-    for (Map.Entry<String, String> entry : properties.entrySet()) {
-      json = json + "    \"" + entry.getKey() + "\":\"" + entry.getValue() + "\",";
-    }
-    if (json.charAt(json.length() - 1) == ',') {
-      json = json.substring(0, json.length() - 1);
+    if (response.getStatus() == OK) {
+      return response.readEntity(Application.class);
     }
-    json = json + "  }\n";
-    return json;
-  }
 
-  public void updateApplication(ApplicationMetadata metadata) {
-    Entity<String> json = updateApplicationRequest(metadata);
-    Response response = put(baseUri, String.format(APPLICATION_UPDATE_PATH, metadata.getName()), json);
-    if (response.getStatus() != OK && response.getStatus() != MOVED_PERMANENTLY) {
+    if (response.getStatus() == NOT_FOUND) {
```

Comment:
This class looks much better now. If you want a suggestion, you could throw javax.ws.rs.core.Response.NotFoundException or something like that.